amundsen_application/api/utils/metadata_utils.py

```diff
@@ -1,73 +1,44 @@
-from typing import Dict
-from flask import current_app as app
+from typing import Any, Dict
 
-from amundsen_application.models.user import load_user, dump_user
+from amundsen_common.models.popular_table import PopularTable, PopularTableSchema
+from amundsen_common.models.table import Table, TableSchema
+from flask import current_app as app
 
 
-def marshall_table_partial(table: Dict) -> Dict:
+def marshall_table_partial(table_dict: Dict) -> Dict:
     """
     Forms a short version of a table Dict, with selected fields and an added 'key'
     :param table: Dict of partial table object
     :return: partial table Dict
 
     TODO - Unify data format returned by search and metadata.
     """
-    table_name = table.get('table_name', '')
-    schema_name = table.get('schema', '')
-    cluster = table.get('cluster', '')
-    db = table.get('database', '')
-    return {
-        'cluster': cluster,
-        'database': db,
-        'description': table.get('table_description', ''),
-        'key': '{0}://{1}.{2}/{3}'.format(db, cluster, schema_name, table_name),
-        'name': table_name,
-        'schema_name': schema_name,
-        'type': 'table',
-        'last_updated_epoch': table.get('last_updated_epoch', None),
-    }
+    schema = PopularTableSchema(strict=True)
+    # TODO: consider migrating to validate() instead of roundtripping
+    table: PopularTable = schema.load(table_dict).data
+    results = schema.dump(table).data
+    # TODO: fix popular tables to provide these? remove if we're not using them?
+    results['last_updated_timestamp'] = None
+    results['type'] = 'table'
```

Comment:
We actually need to preserve the `key` here
```
    # TODO: Add the 'key' or 'id' to the base PopularTableSchema
    results['key'] = '{0}://{1}.{2}/{3}'.format(table.db, table.cluster, table.schema_name, table.table_name)
```

Reply:
@friendtocephalopods FYI, this is currently breaking UI and blocker of `v2.0.0`

Reply:
Oh! Good catch! I haven't quite pushed 'key' as an attribute to the common models yet. Let me replace the key...